### PR TITLE
[WIFI-9985] Add CORS support

### PR DESCRIPTION
**Pull request by elludraon**
Add proper CORS support to `ApiServer`.
- Add global HTTP `OPTIONS` handler to reflect requested options.
- Add "Access-Control-Allow-Origin" response header (and "Vary") whenever "Origin" request header is present, *if* CORS checks pass based on `APISERVERPARAMS_CORSDOMAINLIST` RRM config:
    - If set, origin must match a domain in this comma-separated list.
    - If empty, always reflect the "Origin" header (i.e. allow all origins).

### CODE REVIEW

File: src/main/java/com/facebook/openwifirrm/modules/ApiServer.java [hunk 1]
```diff
@@ -338,6 +362,26 @@ private void beforeFilter(Request request, Response response) {
...
+		String origin = request.headers("Origin");
+		if (origin != null) {
+			boolean allowOrigin = false;
+			if (corsDomains == null) {
+				allowOrigin = true;
+			} else {
+				try {
+					String originHost = new URI(origin).getHost();
+					allowOrigin = corsDomains.contains(originHost);
+				} catch (URISyntaxException e) { /* invalid origin URI */ }
```
Comment (RockyMandayam2) [inline src/main/java/com/facebook/openwifirrm/modules/ApiServer.java:375]: Should there be some kind of log message here?

File: src/main/java/com/facebook/openwifirrm/modules/ApiServer.java [hunk 2]
```diff
@@ -372,6 +416,22 @@ private void afterFilter(Request request, Response response) {
...
+		String acrh = request.headers("Access-Control-Request-Headers");
+		if (acrh != null) {
+			response.header("Access-Control-Allow-Headers", acrh);
+		}
+		String acrm = request.headers("Access-Control-Request-Method");
+		if (acrm != null) {
+			response.header("Access-Control-Allow-Methods", acrm);
+		}
+		response.header("Access-Control-Allow-Credentials", "true");
+		response.header("Access-Control-Max-Age", "86400");
```
Comment (RockyMandayam2) [inline src/main/java/com/facebook/openwifirrm/modules/ApiServer.java:431]: Maybe make 86400 a parameter

Comment (RockyMandayam2) [inline src/main/java/com/facebook/openwifirrm/modules/ApiServer.java:431]: Maybe add a comment saying it is 86400 seconds -> 24 hours.

**Review by RockyMandayam2 — APPROVED** [review]

File: src/main/java/com/facebook/openwifirrm/modules/ApiServer.java (hunk 1 above)
Comment (elludraon) [reply src/main/java/com/facebook/openwifirrm/modules/ApiServer.java:375]: This gets captured in the "CORS checks failed" log below

File: src/main/java/com/facebook/openwifirrm/modules/ApiServer.java (hunk 2 above)
Comment (elludraon) [reply src/main/java/com/facebook/openwifirrm/modules/ApiServer.java:431]: I decided not to since this isn't important to change, it's just the cache time

Comment (elludraon) [reply src/main/java/com/facebook/openwifirrm/modules/ApiServer.java:431]: OK

**Pull request merged**